task.sql

```diff
@@ -2,0 +1,35 @@
+CREATE DATABASE IF NOT EXISTS ShopDB;
+
+USE ShopDB;
+
+CREATE TABLE IF NOT EXISTS Products (
+    ID INT AUTO_INCREMENT PRIMARY KEY,
+    Name VARCHAR(50) NOT NULL,
+    Description VARCHAR(100),
+    Price DECIMAL(10, 2) NOT NULL,
+    WarehouseAmount INT NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS Customers (
+    ID INT AUTO_INCREMENT PRIMARY KEY,
+    FirstName VARCHAR(50) NOT NULL,
+    LastName VARCHAR(50) NOT NULL,
+    Email VARCHAR(100) NOT NULL,
+    Address VARCHAR(100)
+);
+
+CREATE TABLE IF NOT EXISTS Orders (
+    ID INT AUTO_INCREMENT PRIMARY KEY,
+    CustomerID INT,
+    Date DATE NOT NULL,
+    FOREIGN KEY (CustomerID) REFERENCES Customers(ID) ON DELETE SET NULL
+);
+
+CREATE TABLE IF NOT EXISTS OrderItems (
+    ID INT AUTO_INCREMENT PRIMARY KEY,
+    OrderID INT,
```

Comment:
The `OrderID` column should allow NULL values since the foreign key constraint uses `ON DELETE SET NULL`. Add `NULL` to the column definition to avoid errors.